gobblin-utility/src/main/java/gobblin/util/JobLauncherUtils.java

```diff
@@ -114,6 +114,39 @@ public static void cleanStagingData(List<? extends State> states, Logger logger)
   }
 
   /**
+   * Cleanup staging data of all tasks of a job.
+   *
+   * @param state job state
+   */
+  public static void cleanJobStagingData(State state, Logger logger) throws IOException {
+    Preconditions.checkArgument(state.contains(ConfigurationKeys.WRITER_STAGING_DIR),
+        "Missing required property " + ConfigurationKeys.WRITER_STAGING_DIR);
+    Preconditions.checkArgument(state.contains(ConfigurationKeys.WRITER_OUTPUT_DIR),
+        "Missing required property " + ConfigurationKeys.WRITER_OUTPUT_DIR);
+
+    String writerFsUri = state.getProp(ConfigurationKeys.WRITER_FILE_SYSTEM_URI, ConfigurationKeys.LOCAL_FS_URI);
+    FileSystem fs = getFsWithProxy(state, writerFsUri);
+
+    Path jobStagingPath = new Path(state.getProp(ConfigurationKeys.WRITER_STAGING_DIR));
+    logger.info("Cleaning up staging directory " + jobStagingPath);
+    HadoopUtils.deletePath(fs, jobStagingPath, true);
+
+    if (fs.listStatus(jobStagingPath.getParent()).length == 0) {
+      logger.info("Deleting directory " + jobStagingPath.getParent());
+      HadoopUtils.deletePath(fs, jobStagingPath.getParent(), true);
```

Comment:
Each run will create a `task-data/[jobId]/task-staging` dir and `task-data/[jobId]/task-output` dir. I need to delete the parent of task staging and task output (i.e `task-data/[jobId]`) to avoid creating an empty dir in each run.

And since this structure is not enforced (one may set `writer.staging.dir` and `writer.output.dir` to any dir even though they are deprecated), I can't return here because I still need to check whether task output dir is deleted. I also can't just delete `task-data/[jobId]` for this reason.